alphabase/statistics/regression.py

```diff
@@ -0,0 +1,235 @@
+# AUTOGENERATED! DO NOT EDIT! File to edit: ../../nbdev_nbs/statistics/regression.ipynb.
+
+# %% auto 0
+__all__ = ['EPSILON', 'LOESSRegression']
+
+# %% ../../nbdev_nbs/statistics/regression.ipynb 2
+import numpy as np
+
+from sklearn.preprocessing import PolynomialFeatures
+from sklearn.base import BaseEstimator, RegressorMixin
+
+from sklearn.utils.estimator_checks import check_estimator
+import matplotlib.pyplot as plt
+
+# %% ../../nbdev_nbs/statistics/regression.ipynb 3
+EPSILON = 1e-6
+
+class LOESSRegression(BaseEstimator, RegressorMixin):
+
+    def __init__(self, 
+                n_kernels: int = 6, 
+                kernel_size: float = 2., 
+                polynomial_degree: int = 2):
+        """scikit-learn estimator which implements a LOESS style local polynomial regression.
+        The number of basis functions or kernels can be explicitly defined which allows for faster and cheaper training and inference.
+        
+        Parameters
+        -----------
+
+        n_kernels: int, default = 6
+            The number of local polynomial functions used to approximate the data. The location and extend of the kernels will be distributed to contain an equal number of datapoints in the training set.
+
+        kernel_size: float, default = 2
+            A factor increasing the kernel size to overlap with the neighboring kernel.
+
+        polynomial_degree: int, default = 2
+            Degree of the polynomial functions used for the local approximation.
+
+        """
+        self.n_kernels = n_kernels
+        self.kernel_size = kernel_size
+        self.polynomial_degree = polynomial_degree
+
+    def get_params(self, deep: bool = True):
+        return super().get_params(deep)
+
+    def set_params(self, **params):
+        return super().set_params(**params)
+
+    def _more_tags(self):
+        return {'X_types': ['1darray']}
+
+    def calculate_kernel_indices(self, x: np.ndarray):
+        """Determine the indices of the datapoints belonging to each kernel.
+
+        Parameters
+        -----------
+        x : numpy.ndarray, float, of shape (n_datapoints)
```

Comment:
For nbdev2, it is better to use:

```
x : np.ndarray
    float, of shape (n_datapoints)
```

Otherwise the contents after 'float' will be hided